lib/ash/type/type.ex

```diff
@@ -541,7 +541,13 @@ defmodule Ash.Type do
       if constraints[:nil_items?] do
         StreamData.one_of([StreamData.constant(nil), generator])
       else
-        generator
+        if type in [Ash.Type.String, :string, Ash.Type.CiString, :ci_string] &&
+             !item_constraints(constraints)[:allow_empty?] do
+          generator
+          |> StreamData.filter(&(&1 != ""))
```

Comment:
I think what we need to do is use `StreamData.filter` to see if the value, when passed through `Ash.Type.apply_constraints` would then be `nil`. That would solve this for all types, instead of just strings. To get the item constraints, use `constraints[:items] || []`